app/helpers/issue_view_columns_issues_helper.rb

```diff
@@ -6,49 +6,238 @@ def render_descendants_tree(issue)
     # no field defined, then use render from core redmine (or whatever by other plugins loaded before this)
     return super if columns_list.count.zero?
 
-    # continue here if there are fields defined
+    # Retrieve minimum width settings for columns
+    min_width_setting = RedmineIssueViewColumns.setting(:columns_min_width)
+    min_widths = {}
+    if min_width_setting.present?
+      min_width_setting.split(",").each do |column_setting|
+        column_name, min_width = column_setting.split(":").map(&:strip)
+        min_widths[column_name] = min_width
+      end
+    end
+
+    # Retrieve sorting settings and determine if sorting by directory/file model is enabled
+    sort_dir_file_model = RedmineIssueViewColumns.setting(:sort_dir_file_model)
+    collapsed_ids = issue.collapsed_ids.to_s.split.map(&:to_i)
     field_values = +''
     s = table_start_for_relations columns_list
     manage_relations = User.current.allowed_to? :manage_subtasks, issue.project
-    # set data
-    issue_list(issue.descendants.visible.preload(:status, :priority, :tracker, :assigned_to).sort_by(&:lft)) do |child, level|
```

Comment:
It's not because we still need to render the issues recursevely because the hidden attribute is passed down from the parent when you collapse. Also ,"sort by" takes only one arg and it's complicated making a custom function that takes both lft and our sorting criteria in consideration .Since lft doesn't hold information about the level we are on , we need to additionally check for both issues compared to have the same parent.